memory/userspace_mm.log

```diff
@@ -0,0 +1,240 @@
+Starting malloc() test...
+malloc: 2^0: 2266 ns
+free: 2^0: 1554 ns
+malloc: 2^1: 387 ns
+free: 2^1: 292 ns
+malloc: 2^2: 168 ns
+free: 2^2: 117 ns
+malloc: 2^3: 149 ns
+free: 2^3: 163 ns
+malloc: 2^4: 250 ns
+free: 2^4: 191 ns
+malloc: 2^5: 760 ns
+free: 2^5: 281 ns
+malloc: 2^6: 332 ns
+free: 2^6: 195 ns
+malloc: 2^7: 403 ns
+free: 2^7: 185 ns
+malloc: 2^8: 277 ns
+free: 2^8: 208 ns
+malloc: 2^9: 378 ns
+free: 2^9: 123 ns
+malloc: 2^10: 410 ns
+free: 2^10: 177 ns
+malloc: 2^11: 9652 ns
+free: 2^11: 1212 ns
+malloc: 2^12: 354 ns
+free: 2^12: 381 ns
+malloc: 2^13: 7120 ns
+free: 2^13: 228 ns
+malloc: 2^14: 7500 ns
+free: 2^14: 333 ns
+malloc: 2^15: 7052 ns
+free: 2^15: 268 ns
+malloc: 2^16: 7065 ns
+free: 2^16: 253 ns
+malloc: 2^17: 5813 ns
+free: 2^17: 159 ns
+malloc: 2^18: 17328 ns
+free: 2^18: 22104 ns
+malloc: 2^19: 9915 ns
+free: 2^19: 9326 ns
+malloc: 2^20: 8767 ns
+free: 2^20: 8427 ns
+malloc: 2^21: 12057 ns
+free: 2^21: 14842 ns
+malloc: 2^22: 12054 ns
+free: 2^22: 12178 ns
+malloc: 2^23: 10963 ns
+free: 2^23: 12049 ns
+malloc: 2^24: 10441 ns
+free: 2^24: 11734 ns
+malloc: 2^25: 10362 ns
+free: 2^25: 11815 ns
+malloc: 2^26: 11766 ns
+free: 2^26: 13536 ns
+malloc: 2^27: 11446 ns
+free: 2^27: 12738 ns
+malloc: 2^28: 10299 ns
+free: 2^28: 14192 ns
+malloc: 2^29: 10561 ns
+free: 2^29: 17027 ns
+malloc: 2^30: 15088 ns
+free: 2^30: 24716 ns
+malloc: 2^31: 13175 ns
+free: 2^31: 23985 ns
+malloc: 2^32: 12708 ns
+free: 2^32: 32658 ns
+malloc: 2^33: 13759 ns
+free: 2^33: 24580 ns
+Stopped at 17179869184 bytes,34 power
+malloc() test finished.
+
+Starting calloc() test #1...
+calloc: 2^0: 2580 ns
+free: 2^0: 602 ns
+calloc: 2^1: 566 ns
+free: 2^1: 168 ns
+calloc: 2^2: 353 ns
+free: 2^2: 138 ns
+calloc: 2^3: 330 ns
+free: 2^3: 110 ns
+calloc: 2^4: 372 ns
+free: 2^4: 164 ns
+calloc: 2^5: 384 ns
+free: 2^5: 145 ns
+calloc: 2^6: 476 ns
+free: 2^6: 156 ns
+calloc: 2^7: 946 ns
+free: 2^7: 134 ns
+calloc: 2^8: 331 ns
+free: 2^8: 129 ns
+calloc: 2^9: 560 ns
+free: 2^9: 137 ns
+calloc: 2^10: 481 ns
+free: 2^10: 125 ns
+calloc: 2^11: 735 ns
+free: 2^11: 567 ns
+calloc: 2^12: 933 ns
+free: 2^12: 470 ns
+calloc: 2^13: 11626 ns
+free: 2^13: 323 ns
+calloc: 2^14: 8994 ns
+free: 2^14: 361 ns
+calloc: 2^15: 26205 ns
+free: 2^15: 302 ns
+calloc: 2^16: 50738 ns
+free: 2^16: 175 ns
+calloc: 2^17: 117141 ns
+free: 2^17: 225 ns
+calloc: 2^18: 233444 ns
+free: 2^18: 371 ns
+calloc: 2^19: 53405 ns
+free: 2^19: 353 ns
+calloc: 2^20: 737183 ns
+free: 2^20: 713 ns
+calloc: 2^21: 972546 ns
+free: 2^21: 539 ns
+calloc: 2^22: 1968626 ns
+free: 2^22: 1704 ns
+calloc: 2^23: 4126972 ns
+free: 2^23: 1218 ns
+calloc: 2^24: 8807862 ns
+free: 2^24: 1986 ns
+calloc: 2^25: 32726 ns
```

Comment:
Do You have an idea why allocation of 2^25 takes less time compare to allocation of 2^24 bytes?

Reply:
@yekovalyov I guess when it was trying to allocate 2^24, the buffer that had been previously allocated was not big enough, and the calloc issued a system call to allocate more memory. Since big enough buffer was already allocated, it didn't need to make a system call when allocating 2^25.